daffodil-core/src/main/scala/org/apache/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -625,6 +630,16 @@ trait ElementBaseGrammarMixin
     }
   }
 
+  private lazy val textStandardNonBaseTenConverter = {
+    primType match {
+      case PrimType.Double | PrimType.Float | PrimType.Decimal =>
+        SDE("dfdl:textStandardBase=\"%s\" cannot be used with %s", textStandardBaseDefaulted, primType.globalQName)
+      case _: NodeInfo.Numeric.Kind => ConvertNonBaseTenTextNumberPrim(this)
+      case PrimType.HexBinary | PrimType.Boolean | PrimType.Date | PrimType.Time | PrimType.DateTime | PrimType.AnyURI | PrimType.String =>
```

Comment:
Can this be a wildcard match instead of listing every type?

Reply:
Yeah, that's probably the right thing to do in this case. I usually like to list everything because scala will warn you if the match/case isn't exhaustive. So it sort of forces you to think about every case. And if we add a new type (e.g. like we did recently for blobs) this will error and we'll be forced to think about how this new type applies to this match. But in this case I'm not sure if it's worth it.